string expected
---
int8 num_samples    